packages/flutter/lib/src/material/data_table.dart

```diff
@@ -851,15 +895,47 @@ class DataTable extends StatelessWidget {
       ?? dataTableTheme.dataRowHeight
       ?? themeData.dataTableTheme.dataRowHeight
       ?? kMinInteractiveDimension;
-    label = Container(
-      padding: padding,
-      height: effectiveDataRowHeight,
-      alignment: numeric ? Alignment.centerRight : AlignmentDirectional.centerStart,
-      child: DefaultTextStyle(
-        style: effectiveDataTextStyle.copyWith(
-          color: placeholder ? effectiveDataTextStyle.color!.withOpacity(0.6) : null,
+
+    String semanticsLabel = '';
+    if (rowHeading != null) {
+      semanticsLabel += '$rowHeading, ';
```

Comment:
just curious, how would visually impaired user know what exactly in the cell if we announce headings and cell content.